bootstrap/kubeadm/internal/controllers/kubeadmconfig_controller.go

```diff
@@ -991,12 +993,19 @@ func (r *KubeadmConfigReconciler) reconcileTopLevelObjectSettings(ctx context.Co
 	}
 }
 
-// storeBootstrapData creates a new secret with the data passed in as input,
+func (r *KubeadmConfigReconciler) getSecret(ctx context.Context, namespace string, secretName string) (*corev1.Secret, error) {
+	bootstrapDataSecret := &corev1.Secret{}
+	if err := r.Client.Get(ctx, types.NamespacedName{Namespace: namespace, Name: secretName}, bootstrapDataSecret); err != nil {
```

Comment:
We should probably use a live client here instead of the normal cached one?

Reply:
@vincepri Thanks for the review. 
IIUC, from here , https://github.com/kubernetes-sigs/cluster-api/blob/main/bootstrap/kubeadm/main.go#L179 , I see the  ClientDisableCacheFor added for &corev1.Secret{} . So it would directly get from apiserver?
Is the live client still needed here? or i totally misunderstood the intent here?

Reply:
Good catch, sounds like it was already taken care of!